lib/layout/force-directed-layout.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * force-directed-layout.js
+ *
+ * This module contains logic for performing the force-directed layout
+ * algorithm on a graph.
+ *
+ * @author John O'Brien.
+ */
+ 
+ 
+
+var Graph = require('./../model/graph');
+var Vertex = require('./../model/vertex');
+var Port = require('./../model/port');
+var Edge = require('./../model/edge');
+var CardinalPortPosition = require('./../model/cardinalPortPosition');
+var change = false;
+//ideal length
+var iLength = 100;
+//force strength
+var forceStr = 1.0/3.0;
+//stretching force
+var stretchForce  = 0.70;
+var vertices;
+var mGraph;
+
+
+//Based partially on https://code.google.com/p/jung/source/browse/branches/guava/jung/jung-algorithms/src/main/java/edu/uci/ics/jung/algorithms/layout/SpringLayout.java?r=19
+
+function layOutGraph(graph) {
+	mGraph = graph;
+	vertices = graph.get('vertices');
+	var edgeList = graph.get('edges');
+	var runner = true;
+	//var loopCount =0;
+	while(runner){
+		//console.log('loop count is: '+loopCount+'\n');
+		change = false;
+		
+		for(var i=0;i<edgeList.length;i++){
+			springForce(edgeList[i]);
+		}
+		if(change === false){
+			runner = false;
+		}
+		//loopCount++;
+	}
+	//defaulting all ports that do not have a position
+	for(var j=0;j < vertices.length; j++){
+		var inputs = vertices[j].get('inputs');
+		var outputs = vertices[j].get('outputs');
+		
+		for(var h =0; h < inputs.length; h++){
+			
+			if(inputs[h].get('position') === null){
+				inputs[h].set('position', new cardinalPortPosition('E', 50));
+			}
+			
+		}
+		for(h =0;h < outputs.length; h++){
+			if(outputs[h].get('position') === null){
+				outputs[h].set('position', new cardinalPortPosition('W', 50));
+			}
+		}
+	}
+}
+
+function springForce(edge){
```

Comment:
Also also, it would be nice to have a brief comment that describes what this function does. I'm having a hard time understanding it.